spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/KubernetesConfigDataLocationResolver.java

```diff
@@ -167,56 +140,89 @@ protected KubernetesNamespaceProvider kubernetesNamespaceProvider(Environment en
 		return new KubernetesNamespaceProvider(environment);
 	}
 
-	@Override
-	public int getOrder() {
-		return -1;
-	}
+	private void registerProperties(ConfigDataLocationResolverContext resolverContext,
+			KubernetesClientProperties clientProperties, ConfigMapConfigProperties configMapProperties,
+			SecretsConfigProperties secretsProperties) {
 
-	private BindHandler getBindHandler(ConfigDataLocationResolverContext context) {
-		return context.getBootstrapContext().getOrElse(BindHandler.class, null);
-	}
+		ConfigurableBootstrapContext bootstrapContext = resolverContext.getBootstrapContext();
+		registerSingle(bootstrapContext, KubernetesClientProperties.class, clientProperties,
+				"configDataKubernetesClientProperties");
 
-	protected PropertyHolder loadProperties(ConfigDataLocationResolverContext context) {
-		Binder binder = context.getBinder();
-		BindHandler bindHandler = getBindHandler(context);
-		String applicationName = binder.bind("spring.application.name", String.class).orElse(null);
-		String namespace = binder.bind("spring.cloud.kubernetes.client.namespace", String.class)
-				.orElse(binder.bind("kubernetes.namespace", String.class).orElse(""));
-
-		KubernetesClientProperties kubernetesClientProperties;
-		if (context.getBootstrapContext().isRegistered(KubernetesClientProperties.class)) {
-			kubernetesClientProperties = new KubernetesClientProperties();
-			BeanUtils.copyProperties(context.getBootstrapContext().get(KubernetesClientProperties.class),
-					kubernetesClientProperties);
+		if (configMapProperties != null) {
+			registerSingle(bootstrapContext, ConfigMapConfigProperties.class, configMapProperties,
+					"configDataConfigMapConfigProperties");
 		}
-		else {
-			kubernetesClientProperties = binder.bindOrCreate(KubernetesClientProperties.PREFIX,
-					Bindable.of(KubernetesClientProperties.class), bindHandler);
+
+		if (secretsProperties != null) {
+			registerSingle(bootstrapContext, SecretsConfigProperties.class, secretsProperties,
+					"configDataSecretsConfigProperties");
 		}
-		kubernetesClientProperties.setNamespace(namespace);
+	}
 
-		boolean configEnabled = binder.bind("spring.cloud.kubernetes.config.enabled", boolean.class).orElse(true);
-		boolean secretsEnabled = binder.bind("spring.cloud.kubernetes.secrets.enabled", boolean.class).orElse(true);
+	protected record PropertyHolder(KubernetesClientProperties kubernetesClientProperties,
+			ConfigMapConfigProperties configMapConfigProperties, SecretsConfigProperties secretsProperties,
+			String applicationName) {
 
-		ConfigMapConfigProperties configMapConfigProperties = null;
-		if (configEnabled) {
-			configMapConfigProperties = binder.bindOrCreate(ConfigMapConfigProperties.PREFIX,
-					ConfigMapConfigProperties.class);
-		}
+		private static PropertyHolder of(ConfigDataLocationResolverContext context) {
```

Comment:
I moved everything related to configuration properties in here, imho, it simplifies reading the code quite a lot